tests/component/records/datetime_cases.py

```diff
@@ -0,0 +1,43 @@
+SAMPLE_YEAR = 1983
+SAMPLE_YEAR_SHORT = 83
+SAMPLE_MONTH = 1
+SAMPLE_DAY = 2
+
+
+def create_sample(template: str) -> str:
+    return (
+        template
+        .replace('YYYY', str(SAMPLE_YEAR))
+        .replace('YY', str(SAMPLE_YEAR_SHORT))
+        .replace('MM', str(SAMPLE_MONTH))
+        .replace('DD', str(SAMPLE_DAY))
+    )
+
+
+# TODO: Convert back to date cases probably
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker